core/replica.go

```diff
@@ -46,6 +42,32 @@ type replica struct {
 	handleClientStream messageStreamHandler
 }
 
+type options struct {
+	logger *commonLogger.Logger
```

Comment:
`commonLogger.Logger` is an interface, which is a pointer type. What is a point in making `logger` field of a double-pointer type? Could we change it to `logger commonLogger.Logger`? Same applies for the parameter of `WithLogger`.

Reply:
Sometimes my previous C++ experience reminds about it :)